### DISCUSSION TITLE
[Bug] the thrift plugin in javaagent doesn't handle exceptions.

### DISCUSSION THREAD

**Discussion by h0cheung** [apache/skywalking#12300] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

The thrift plugin can't handle exceptions.

[The `ProcessFunction` has caught all exceptions except `TTransportException` for many years.](https://github.com/apache/thrift/commit/026c9d032c4a298ecb9edbcdfb52590facb442f3) So the plugin, which enhances `TBaseProcessor` cannot catch the exceptions. Then thrift server spans will never have errors unless a transport exception happens.

On the client side, the plugin works well. The thrift client spans will have errors when an exception happens.

### What you expected to happen

If an exception happens, both server and client should have errors.

### How to reproduce

Run a thrift server, which throw an custom Exception that will be returned to thrift client.

Then send requests to this server.

### Anything else

[For gRPC, the status code in response will be used to find errors.](https://github.com/apache/skywalking-java/blob/313d7d5c8e171df0a24b273c4949738133586922/apm-sniffer/apm-sdk-plugin/grpc-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/grpc/v1/server/TracingServerCall.java) Maybe we should try to handle the response sent by thrift server.

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
This is a common case, that some plugins are not powerful enough to collect everything.

If you or anyone is willing to enhance the plugin, that is great. 

I am moving this from bug to discussion, as 
1. This doesn't block anyone.
2. Plugin use cases are countless, it relies on the whole community users to collaborate, rather than maintainers' work.
